node/src/accountant/payment_adjuster/mod.rs

```diff
@@ -218,81 +201,105 @@ impl PaymentAdjusterReal {
         }
     }
 
-    fn propose_adjustments_recursively<AR, RT>(
-        &mut self,
-        unresolved_accounts: Vec<WeightedPayable>,
-        adjustment_runner: AR,
-    ) -> RT
-    where
-        AR: AdjustmentRunner<ReturnType = RT>,
-    {
-        diagnostics!(
-            "\nUNRESOLVED QUALIFIED ACCOUNTS IN CURRENT ITERATION:",
-            &unresolved_accounts
-        );
+    fn resolve_initial_adjustment_dispatch(
+        &self,
+        weighed_payables: Vec<WeighedPayable>,
+    ) -> Result<
+        Either<Vec<AdjustedAccountBeforeFinalization>, Vec<PayableAccount>>,
+        PaymentAdjusterError,
+    > {
+        if let Some(limit) = self.inner.transaction_count_limit_opt() {
+            return self.begin_with_adjustment_by_transaction_fee(weighed_payables, limit);
+        }
 
-        adjustment_runner.adjust_accounts(self, unresolved_accounts)
+        Ok(Either::Left(
+            self.propose_possible_adjustment_recursively(weighed_payables),
+        ))
     }
 
     fn begin_with_adjustment_by_transaction_fee(
-        &mut self,
-        weighted_accounts: Vec<WeightedPayable>,
-        already_known_affordable_transaction_count: u16,
+        &self,
+        weighed_accounts: Vec<WeighedPayable>,
+        transaction_count_limit: u16,
     ) -> Result<
         Either<Vec<AdjustedAccountBeforeFinalization>, Vec<PayableAccount>>,
         PaymentAdjusterError,
     > {
-        let error_factory = LateServiceFeeSingleTxErrorFactory::new(&weighted_accounts);
+        diagnostics!(
+            "\nBEGINNING WITH ADJUSTMENT BY TRANSACTION FEE FOR ACCOUNTS:",
+            &weighed_accounts
+        );
 
-        let weighted_accounts_affordable_by_transaction_fee =
-            dump_unaffordable_accounts_by_transaction_fee(
-                weighted_accounts,
-                already_known_affordable_transaction_count,
-            );
+        let error_factory = LateServiceFeeSingleTxErrorFactory::new(&weighed_accounts);
```

Comment:
I hope we already have a plan to shorten the name of this structure. If not please do that... here is a suggestion: `ServiceFeeErrorFactory`.